src/parser/ui/TalentSpellText.tsx

```diff
@@ -0,0 +1,48 @@
+import { SpellIcon, TooltipElement } from 'interface';
+import { SpellLink } from 'interface';
+import CombatLogParser from 'parser/core/CombatLogParser';
+import PropTypes from 'prop-types';
+import { ReactNode } from 'react';
+
+interface Props {
+  spellId: number;
+  maxRanks: number;
+  children: ReactNode;
+  className?: string;
+}
+
+interface Context {
+  parser: CombatLogParser;
+}
+
+/**
+ * Component to display rank/maxrank for a talent. Pass the spell ID and maxranks of the talent in the `spellId` prop.
+ *
+ * Component will link to the talent with the current rank as well".
+ */
+const TalentSpellText = (
+  { spellId, maxRanks, children, className }: Props,
+  { parser: { selectedCombatant } }: Context,
+) => {
+  const rank = selectedCombatant.getTalentRank(spellId);
+  return (
+    <div className={`pad boring-text ${className || ''}`}>
+      <label>
+        <SpellIcon id={spellId} /> <SpellLink id={spellId} icon={false} /> -{' '}
+        <TooltipElement
+            content={`Talented into with ${rank} point${
+              rank > 1 ? 's' : ''
+            } out of ${maxRanks} point${maxRanks > 1 ? 's' : ''}`}
+          >
+            {rank}/{maxRanks}
+          </TooltipElement>
+      </label>
+      <div className="value">{children}</div>
+    </div>
+  );
+};
+TalentSpellText.contextTypes = {
+  parser: PropTypes.object.isRequired,
+};
```

Comment:
An interesting new display, was considering doing something like this myself. Do you have a screenshot of how it looks?

Reply:
It is inspired by the current conduit field 

![image](https://user-images.githubusercontent.com/73188508/195025246-fbc2af9c-080c-4c23-b219-dbf67ab87495.png)

![image](https://user-images.githubusercontent.com/73188508/195025324-8a2b14b8-f8c7-44d5-be57-cf0d1ba3f7d9.png)

Reply:
lgtm

Reply:
Actually looking at this I like the display of the talent points, but I don't agree at all with displaying overheal so prominently - I can't think of anything else in WoWA that shows overheal like that, and I can't see anything special in the talent that makes showing overheal more important.

This isn't enough to block the PR though, I'll merge it in. Should follow up on this afterwards, I'd like to know the thought process.